### ISSUE TITLE
Match position correction to PtychoShelves more closely

### ISSUE THREAD

**Issue by mdw771** [AdvancedPhotonSource/pty-chi#71]
There are still some remaining discrepancies in the position correction routines between Pty-Chi and PtychoShelves. Since PtychoShelves has been used to deliver successful 3D reconstructions, we treat it as the standard in our first steps. The plan is:
1. Create a new branch with the position correction math and default settings more closely matched with PtychoShelves.
2. Compare the new branch and the current main branch on a ptycho-tomographic dataset and see which one gives better 3D resolution.
3. Decide whether to merge or partially merge the changes.

**Comment by mdw771** [comment]
## Summary of changes implemented on position_correction_fix

This section records the final behavior implemented in commits 7df5981,
30a7de4, 4295d46, and 3825c88. The earlier sections preserve the pre-change
comparison that motivated the work.

### 1. Least-squares denominator

**PtychoShelves**

For each active scan position and coordinate, PtychoShelves uses the local
diagonal least-squares solution
```
\[
\delta_{i,a}=\frac{b_{i,a}}{h_{i,aa}},
\]
```
without an additional damping term or an explicit zero-denominator guard.

**Previous Pty-Chi**

Pty-Chi added the maximum directional curvature in the current batch:
```
\[
\delta_{i,a}
=
\frac{b_{i,a}}
{h_{i,aa}+\max(\max_j h_{j,aa},\epsilon)}.
\]
```
This coupled the update scale to the other positions in the minibatch and
reduced every raw correction by at least a factor of two relative to the
undamped quotient.

**Changed Pty-Chi**

Pty-Chi now uses the undamped local quotient when \(h_{i,aa}>0\), matching the
PtychoShelves least-squares construction. When \(h_{i,aa}=0\), it returns zero,
which is the natural pseudoinverse result for an unconstrained coordinate.
A safe denominator is used before torch.where so that no \(0/0\) intermediate
is formed. Because the calculation lives in the shared gradient-correction
routine, this mathematical change applies to every reconstructor using that
routine.

### 2. Default probe-position step size

**PtychoShelves**

PtychoShelves adds the computed position correction directly, so the implicit
outer position step size is 1.

**Previous Pty-Chi**

Pty-Chi passed the correction through the configured position optimizer, whose
shared default learning rate was 0.3.

**Changed Pty-Chi**

The shared ProbePositionOptions.step_size default is now 1.0. This applies to
LSQML, PIE/ePIE/rPIE, DM, RAAR, BH, and autodiff. Under the default plain-SGD
configuration, the optimizer now applies the calculated correction without an
additional scalar reduction.

### 3. Clipping population and ordering

In all three descriptions below,
```
\[
L_a
=
\min\left(
L_{\mathrm{user}},
10\,\operatorname{MAD}_i(\delta_{i,a})
\right)
\]
```
is a per-axis, componentwise clipping threshold.

**PtychoShelves**

PtychoShelves calculates MAD over the active group of scan positions, clips the
raw least-squares correction to \(L_a\), and only then stores or processes the
correction for momentum.

**Previous Pty-Chi**

When LSQML position momentum was enabled, Pty-Chi already clipped the raw active
minibatch before momentum. When momentum was disabled, it instead applied the
optimizer first and clipped the resulting displacement afterward. That
post-optimizer calculation operated on the full position tensor, including
zero gradients for positions outside the active minibatch, so its MAD could
differ from the PtychoShelves active-group MAD.

**Changed Pty-Chi**

LSQML now always computes \(L_a\) from the raw active-minibatch correction and
clips before the optional momentum path. LSQML then disables the generic
post-optimizer clipping step, avoiding a second clipping operation and avoiding
the inactive-position zeros. The generic post-optimizer clipping implementation
is retained for other reconstructors.

### 4. Momentum acceleration

**PtychoShelves**

Position momentum is optional and disabled by default. When enabled,
PtychoShelves stores the clipped correction in history, estimates friction from
correlations between recent corrections, uses a hard-coded acceleration gain of
0.5, and does not clip the accelerated correction again. The final displacement
can therefore exceed \(L_a\).

**Previous Pty-Chi**

Position momentum was also optional and disabled when
momentum_acceleration_gain was zero. Pty-Chi exposed the gain to the user
rather than fixing it at 0.5. Its correlation, friction, velocity, far-field,
and small-update gating logic was already modeled on PtychoShelves. When
momentum was enabled, it clipped before history/momentum and did not apply
post-optimizer clipping.

**Changed Pty-Chi**

The configurable momentum form was intentionally retained: the gain was not
hard-coded, and its default remains zero. Because raw clipping is now
unconditional in LSQML, any enabled momentum path always receives and records
the clipped active-minibatch correction. The accelerated result is still not
clipped again, matching the PtychoShelves ordering while allowing the user to
choose the gain.

### 5. Applying the final correction

**PtychoShelves**

PtychoShelves directly applies
```
\[
\mathbf s_i
\leftarrow
\mathbf s_i+\boldsymbol\delta_i.
\]
```
**Previous Pty-Chi**

Pty-Chi stored \(-\boldsymbol\delta_i\) as a PyTorch gradient and invoked the
position optimizer. Default SGD multiplied the correction by 0.3, and the
non-momentum path then applied post-optimizer clipping.

**Changed Pty-Chi**

Pty-Chi still uses its optimizer abstraction, but default SGD now has step size
1 and LSQML no longer invokes post-optimizer clipping. The default update is
therefore the same direct additive correction as PtychoShelves. A user-selected
adaptive optimizer, optimizer momentum, weight decay, or a non-unit learning
rate intentionally changes this equivalence.

### 6. Intentional non-changes

**PtychoShelves**

For an even number of multislice planes, the default correction layer is the
earlier of the two central planes. PtychoShelves also uses its fixed Fourier
derivative and normally applies its relaxed geometry model.

**Previous Pty-Chi**

Pty-Chi selected the later central plane for an even number of slices, allowed
multiple object-differentiation methods, and disabled its affine position
constraint by default.

**Changed Pty-Chi**

These behaviors were left unchanged. They remain affordable or separately
configurable discrepancies rather than part of the least-squares
position-update alignment.
